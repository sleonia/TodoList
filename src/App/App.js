import React from 'react';
import Header from '../Header/Header';
import TodoItem from '../TodoItem/TodoItem';
import Footer from '../Footer/Footer';

class App extends React.Component {
	render () {
		return (
			<div>
				<Header />
				<TodoItem />
				<Footer />
			</div>
		)
	};
};

export default App;
