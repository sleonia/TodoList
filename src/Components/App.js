import React from 'react'
import Header from './Header'
import Footer from './Footer'
//import AddTodo from '../containers/AddTodo'
//import VisibleTodoList from '../containers/VisibleTodoList'

class App extends React.Component {
	render() {
		return (
				<div>
					<Header />
					{/*<AddTodo />*/}
					{/*<VisibleTodoList />*/}
					<Footer />
				</div>
		)
	}
}

export default App