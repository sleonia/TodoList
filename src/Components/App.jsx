import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import rootReducer from '../Reducers/index';

import Header from './Header';
import Footer from './Footer';
import AddItem from '../Containers/AddItem';
import VisibleTodoList from '../Containers/VisibleTodoList';

const localStorageMiddleware = ({ getState }) => next => action => {
  const res = next(action);
  if ([action.type].includes(res.type)) {
    localStorage.setItem('data', JSON.stringify(getState().todos));
  }
  return res;
};

const store = createStore(rootReducer, applyMiddleware(localStorageMiddleware));

const App = () => (
  <>
    <Provider store={store}>
      <Header />
      <AddItem />
      <VisibleTodoList />
      <Footer />
    </Provider>
  </>
);

export default App;
