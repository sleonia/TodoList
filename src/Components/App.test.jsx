import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import App from './App';

configure({ adapter: new Adapter() })

it("Render App component", function () {
  const AppComponent = renderer.create(
    <App />
  );
  expect(AppComponent).toMatchSnapshot();
});
