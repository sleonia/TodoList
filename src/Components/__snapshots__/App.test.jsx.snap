// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App component 1`] = `
Array [
  <header
    className="navbar"
  >
    <h2>
      Tue Sep 08 2020
    </h2>
  </header>,
  <div>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        value=""
      />
      <button
        type="submit"
      >
        Add Item
      </button>
    </form>
  </div>,
  <div
    className="todo_list"
  >
    <ul />
  </div>,
  <div
    className="footer"
  >
    <span>
      Show: 
    </span>
    <button
      className="link"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      All
    </button>
    <button
      className="link"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Active
    </button>
    <button
      className="link"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Completed
    </button>
  </div>,
]
`;
